coreclient/src/clients/process/process_qs.rs

```diff
@@ -222,84 +225,98 @@ impl CoreUser {
-                            .handle_application_message(
-                                application_message,
-                                ds_timestamp,
-                                sender_client_credential.identity(),
-                            )?,
-                        ProcessedMessageContent::ProposalMessage(proposal) => {
-                            self.handle_proposal_message(txn, &mut group, *proposal)
-                                .await?
-                        }
-                        ProcessedMessageContent::StagedCommitMessage(staged_commit) => {
-                            self.handle_staged_commit_message(
-                                txn,
-                                &mut group,
-                                conversation,
-                                *staged_commit,
-                                aad,
-                                ds_timestamp,
-                                &sender,
-                                &sender_client_credential,
-                                we_were_removed,
-                            )
-                            .await?
-                        }
-                        ProcessedMessageContent::ExternalJoinProposalMessage(_) => {
-                            self.handle_external_join_proposal_message()?
-                        }
-                    };
-
-                    // MLSMessage Phase 3: Store the updated group and the messages.
-                    group.store_update(txn.as_mut()).await?;
-
-                    let conversation_messages =
-                        Self::store_messages(txn, &mut notifier, conversation_id, group_messages)
-                            .await?;
-
-                    Ok((
-                        conversation_messages,
-                        conversation_changed,
-                        conversation_id,
-                        profile_infos,
-                    ))
-                })
-                .await?;
+                // MLSMessage Phase 3: Store the updated group and the messages.
+                group.store_update(txn.as_mut()).await?;
+
+                let conversation_messages =
+                    Self::store_messages(txn, notifier, conversation_id, group_messages).await?;
+
+                Ok((
+                    conversation_messages,
+                    conversation_changed,
+                    conversation_id,
+                    profile_infos,
+                ))
+            })
+            .await?;
+
+        // Send delivery receipts for incoming messages
+        // TODO: Queue this and run the network requests batched together in a background task
+
+        let delivered_receipts = conversation_messages.iter().filter_map(|message| {
+            if let Message::Content(content_message) = message.message()
+                && [Disposition::Render, Disposition::Attachment]
+                    .contains(&content_message.content().nested_part.disposition)
```

Comment:
Nitpick: It feels like this should be a `matches!(...)` call.

Reply:
True. Even better, it can be just done with if let now.